src/github/issue-template.ts

```diff
@@ -0,0 +1,126 @@
+import * as path from "path";
+import { GitHub } from "./github";
+import { Component } from "../component";
+import { TextFile } from "../textfile";
+import { YamlFile } from "../yaml";
+
+export interface ContactLinks {
+  /**
+   * The name of the contact link.
+   */
+  readonly name: string;
+
+  /**
+   * The URL of the contact link.
+   */
+  readonly url: string;
+
+  /**
+   * A brief description of the contact link.
+   */
+  readonly about: string;
+}
+
+/**
+ * Options for configuring issue templates in a repository.
+ *
+ * @interface IssueTemplateConfigOptions
+ * @see {@link https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser}
+ */
+export interface IssueTemplateConfigOptions {
+  /**
+   * Indicates whether blank issues (issues without a template) are allowed.
+   *
+   * @type {boolean} [blankIssuesEnabled=false]
+   */
+  readonly blankIssuesEnabled?: boolean;
+
+  /**
+   * An array of contact links to display in the issue template chooser.
+   *
+   */
+  readonly contactLinks?: ContactLinks[];
+}
+
+/**
+ * Options for `IssueTemplate`.
+ */
+export interface IssueTemplateOptions {
+  /**
+   * An array of issue template file names and their contents.
+   * The file names should include the extension (.md or .yml).
+   */
+  readonly templates: { [fileName: string]: string };
+
+  /**
+   * The path within the repository where the issue templates should be created.
+   * Defaults to '.github/ISSUE_TEMPLATE'.
+   */
+  readonly templatePath?: string;
+
+  /**
+   * Whether to include a config.yml file for configuring the issue template chooser.
+   * Defaults to false.
+   */
+  readonly includeConfig?: boolean;
+
+  /**
+   * Configuration options for the config.yml file.
+   * Only applicable if includeConfig is true.
+   */
+  readonly configOptions?: IssueTemplateConfigOptions;
+}
+
+/**
+ * Generates Issue Templates for GitHub repositories
+ * see  {@link https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests}
+ */
+export class IssueTemplate extends Component {
+  constructor(github: GitHub, options: IssueTemplateOptions) {
+    super(github.project);
+    const project = github.project;
```

Comment:
We have moved on from this approach. This is the current state of the art:
```suggestion
  constructor(scope: IScope, options: IssueTemplateOptions) {
    super(scope, "<enter_unique_id_here_if_this_is_a_singleton_component>");
    const project = this.project;
    const github = GitHub.of(project);
    if (!github) {
      // add warning here. Or hard fail, but it seems like the github component is not actually needed for this
    }
```

Reply:
also updated this. thanks for the tip.